xpart/matched_gaussian.py

```diff
@@ -111,18 +111,127 @@ def generate_matched_gaussian_bunch(num_particles,
+                           line=line,
+                           zeta=zeta, delta=delta,
+                           x_norm=x_norm, px_norm=px_norm,
+                           y_norm=y_norm, py_norm=py_norm,
+                           nemitt_x=nemitt_x, nemitt_y=nemitt_y,
+                           weight=total_intensity_particles / num_particles,
+                           **kwargs)
     return part
+
+
+def split_scheme(filling_scheme, communicator=None):
+    """
+    Distribute the filling scheme between the processes, i.e. assign to each
+    processor its bunches
+    """
+    total_n_bunches = len(filling_scheme.nonzero()[0])
+    if communicator is not None:
+        n_procs = communicator.Get_size()
+
+        # create the array containing the id of the bunches on each rank
+        # (copied from PyHEADTAIL.mpi.mpi_data)
+        n_bunches = total_n_bunches
+        n_bunches_on_rank = [n_bunches//n_procs+1 if i < n_bunches % n_procs
+                             else n_bunches // n_procs + 0
+                             for i in range(n_procs)]
+        n_tasks_cumsum = np.insert(np.cumsum(n_bunches_on_rank), 0, 0)
+        total_bunch_ids = np.unique(
+            np.cumsum(filling_scheme == 1)) - 1
+        bunches_per_rank = [total_bunch_ids[n_tasks_cumsum[i]:
+                                            n_tasks_cumsum[i + 1]]
+                            for i in range(n_procs)]
+    else:
+        bunches_per_rank = [np.linspace(0,
+                                        total_n_bunches - 1,
+                                        total_n_bunches)]
+
+    bunches_this_rank = bunches_per_rank[communicator.Get_rank()]
+
+    return bunches_this_rank[0], len(bunches_this_rank)
+
+
+def generate_matched_gaussian_beam(filling_scheme,
+                                   num_particles,
+                                   nemitt_x, nemitt_y, sigma_z,
+                                   total_intensity_particles=None,
+                                   particle_on_co=None,
+                                   R_matrix=None,
+                                   circumference=None,
+                                   momentum_compaction_factor=None,
+                                   rf_harmonic=None,
+                                   rf_voltage=None,
+                                   rf_phase=None,
+                                   p_increment=0.,
+                                   tracker=None,
+                                   line=None,
+                                   particle_ref=None,
+                                   particles_class=None,
+                                   engine=None,
+                                   _context=None, _buffer=None, _offset=None,
+                                   n_bunches=None, first_bunch=None,
+                                   bunch_spacing_buckets=1,
+                                   **kwargs,  # Passed to build_particles
+                                   ):
+
```

Comment:
Docstring (let's write it together, copilot helps :-))

Reply:
At this point this does not work with LineSegmentMap because the function '_characterize_line' fails to infer the harmonic number from it (which is used to define the bucket_length).

Reply:
Now it works with the line segment map as well